### ISSUE TITLE
Refactor unsafe lifecycle methods

### ISSUE THREAD

**Issue by cdock1029** [jamesmfriedman/rmwc#347]
## RMWC or MDC?
> RMWC

### What RMWC Version are you using?
> 3.0.5

### What build system are you using?
> `create-react-app`

### What are the steps to reproduce the bug? 

Render app in `Strict Mode` as described [here](https://reactjs.org/docs/strict-mode.html)

https://codesandbox.io/s/9yp2zjwx4w

### What is the expected behavior?

No console errors

### What is the actual behavior?

```
Warning: Unsafe lifecycle methods were found within a strict-mode tree:
    in StrictMode (created by App)
    in App

componentWillReceiveProps: Please update the following components to use static getDerivedStateFromProps instead: Ripple

Learn more about this warning here:
https://fb.me/react-strict-mode-warnings
```

### Any other information you believe would be useful?

https://reactjs.org/docs/strict-mode.html

**Comment by jamesmfriedman** [comment]
Thanks. I’ll keep this open until they’re gone.

**Comment by jamesmfriedman** [comment]
Fixed in the 5.2.0-alpha.0
